### ISSUE TITLE
Linger option not taken into account 

### ISSUE THREAD

**Issue by abiallais** [nim-lang/nim-zmq#47]
With nim-zmq, the linger option is set when the socket is closed but it seems that the option is not taken into account. 

Environment
nim:  v2.0.2
libzmq: 4.3.4-2 
nim-zmq: v1.5.2

In the following code, the data is never received by the consumer: 

PUSH/PULL pattern
Producer - PUSH
```nim
import std/[os, times]
import zmq

proc main() =
  var sock = connect("tcp://192.168.12.120:33405", PUSH)
  var msg = newString(400 * 1024 * 1024)
  sock.send(msg)
  # set linger to 5s 
  sock.close(5000)

main()
```

Consumer (in python) - PULL
```python
import zmq
ctx = zmq.Context()
sock = ctx.socket(zmq.PULL)
sock.bind("tcp://*:33405)

while True:
  msg = sock.recv_string()
  print(len(msg))
```

The code is working with one of the following modifications, which leads to a linger option issue:
- add a `sleep(5000)` before closing the socket
- replace the main proc with this version
```nim
proc main() =
  var sock = connect("tcp://192.168.12.120:33405", PUSH)
  var msg = newString(400 * 1024 * 1024)
  sock.setsockopt(LINGER, 5000.cint)
  sock.send(msg)
  sock.close()
```
and remove in the connections.nim file (from the nim-zmq library) the following line `setsockopt(c, LINGER, linger.cint)` in the proc close(c: var ZConnectionImpl, linger: int = 500)

**Comment by Clonkk** [comment]
I believe this is a simple typo here : https://github.com/nim-lang/nim-zmq/blob/6eb2f6e4c7b83fd40991a169408bccadf3af4cfc/zmq/connections.nim#L326

The extra linger parameters is not passed and thus the default value is always used

**Comment by Clonkk** [comment]
Making a fix here https://github.com/nim-lang/nim-zmq/issues/47

**Comment by Clonkk** [comment]
This should be fixed in v1.5.3

**Comment by abiallais** [comment]
It is indeed fixed in v1.5.3.
